File: PHP/Classes/Point/Point.php
<?php
    class Point {
        private $x, $y;
        
        private function validate($value) {
            if ( is_numeric($value) ) {
                return $value;
            }
            throw new Exception("Invalid Param");
        }
        
        public function __construct($x=0, $y=0) {
            $this->x = $this->validate($x);
            $this->y = $this->validate($y);
        }
        
        public function __get($name) {
            if ( property_exists($this, $name) ) {
                return $this->$name;
            }
            throw new Exception("Atribute error: atribute $name not found");
        }
        
        public function __set($name, $value) {
            if ( property_exists($this, $name) ) {
                $this->$name = $this->validate($value);
            } else {
                throw new Exception("Atribute error: atribute $name not found");
            }
        }
        
        public function distance(Point $other) {
            return hypot($this->x - $other->x, $this->y - $other->y);
        }
        
        public function __toString() {
            return sprintf("(%g, %g)", $this->x, $this->y);
        }
    }
    
    function compare($x, $y) {
        if ( $x == $y ) {
            echo $x . " is equal to " . $y . PHP_EOL;
        } else {
            echo $x . " is not equal to " . $y . PHP_EOL;
        }
    }
    
    function isTheSameInstance($x, $y) {
        if ( $x === $y ) {
            echo $x . " and " . $y . " is references to the same instance" . PHP_EOL;
        } else {
            echo $x . " and " . $y . " is different instances" . PHP_EOL;
        }
    }
?>

File: cpp/Army/Location/Point.cpp
#include "Point.h"

Point::Point (int x, int y) {
    this->x = x;
    this->y = y;
}

Point::~Point() {}

int Point::getX() const {
    return this->x;
}

int Point::getY() const {
    return this->y;
}

void Point::setX(int value) {
    this->x = value;
}

void Point::setY(int value) {
    this->y = value;
}

double Point::distance(const Point& other) const {
    return hypot(other.x - this->x, other.y - this->y);
}

bool Point::operator==(const Point& other) const {
    return this->x == other.x && this->y == other.y;
}

bool Point::operator!=(const Point& other) const {
    return !(*this == other);
}

bool Point::operator<(const Point& other) const {
    return this->x < other.x || (this->x == other.x && this->y < other.y);
}

std::ostream& operator<<(std::ostream& out, const Point& point) {
    out << "(" << point.getX() << "," << point.getY() << ")";
    return out;
}
